src/bespokelabs/curator/request_processor/base_online_request_processor.py

```diff
@@ -329,39 +329,48 @@ async def process_requests_from_file(
                 await asyncio.gather(*pending_requests)
 
             # Process any remaining retries in the queue
-            pending_retries = []
             while not queue_of_requests_to_retry.empty():
```

Comment:
is there a cleaner way to do this than having a double "while not queue_of_requests_to_retry.empty()" loop? i feel like we should be able to flatten this